arangod/Aql/AqlItemBlock.h

```diff
@@ -320,7 +358,7 @@ class AqlItemBlock {
   /// setValue above puts values in the map and increases the count if they
   /// are already there, eraseValue decreases the count. One can ask the
   /// count with valueCount.
-  std::unordered_map<AqlValue, uint32_t> _valueCount;
```

Comment:
but isn't every AqlValue different now? Do AqlValues refer to external memory that is reference counted? If so why not just store `void*` and remove this `AqlValue::SimpleHash` stuff which is not general purpose anyway

Reply:
AqlValues can refer to inline VelocyPack data or to externally managed memory.
In the former case, the AqlValue is self-contained and doesn't need reference counting.
In the latter case we need it, so that the dynamic memory is freed at the right time.

AqlValues pointing to different dynamic memory locations will be considered different, but each AqlItemBlock may contain the same values multiple times, because values can be copied from one row to the next. That's why we still need the reference counting. And `_valueCount` is exactly where the reference counting happens and where the simplified hash/compare functions are used.
So I don't see a good way of getting rid of this. Any concrete suggestions?

Reply:
I edited the comment not sure if you read that. Why not use void*

Reply:
If I understand correctly, the suggestion is change `_valueCount` from
```
std::unordered_map<AqlValue, ValueInfo> _valueCount
```
to
```
std::unordered_map<void const*, ValueInfo> _valueCount
```
Yes, I think that would work and require less storage space. Obviously it requires changes in all places in AqlItemBlock.* that access `_valueCount`, but it looks pretty doable.
I would like to get this PR here through first, because it should already be an improvement, and then create another PR on top of it to try your suggestion.